hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -309,6 +309,12 @@ def do_draw(self, data):
         reference = data.draw(self.__reference_strategy)
         return machine.names_to_values[reference.name]
 
+    def available(self, data):
+        # ``self_strategy`` is a constant, so drawing from it doesn't count
```

Comment:
The relevant fact is not that `self_strategy` is a constant; it's that `self_strategy` is an instance of the `st.runner()` strategy.  "Drawing" thus returns the current StateMachine without actually touching the underlying buffer.

Reply:
fix by a049b84d5ae31fba0b56440f08e3f790d83c58a2